lib/iris/fileformats/netcdf/saver.py

```diff
@@ -1024,15 +1027,32 @@ def _add_aux_coords(self, cube, cf_var_cube, dimension_names):
             Names associated with the dimensions of the cube.
 
         """
+        from iris.experimental.ugrid.mesh import (
+            Mesh,
+            MeshEdgeCoords,
+            MeshFaceCoords,
+            MeshNodeCoords,
+        )
+
         # Exclude any mesh coords, which are bundled in with the aux-coords.
-        aux_coords_no_mesh = [
+        coords_to_add = [
             coord for coord in cube.aux_coords if not hasattr(coord, "mesh")
         ]
+
+        # Include any relevant mesh location coordinates.
+        mesh: Mesh = getattr(cube, "mesh")
```

Comment:
I'm not really clear why you are including type hints here.  Are you sure it is a good idea?
I'm a bit ambivalent -- not sure that it adds anything much in this case.

Actually I have generally only seen them used  around object properties and interfaces, not in general code.

Also BTW the `location_coords: MeshNodeCoords | MeshEdgeCoords | MeshFaceCoords = getattr( ...` really surprised me - I didn't even know you can use "|" like this, up to now I always used "typing.Union".

Reply:
I wondered if this might be controversial. I agree it offers minimal value to humans reading the code, but that wasn't my motivation:

I'm a big fan of IDE code completion to speed up development, versus manually looking up the properties of any given object. The results of `getattr()` do not get this code completion, because the IDE can't be certain what is being returned, so I'm in the habit of using the hints to access this feature.

I don't know if you consider this a valid reason. I also don't know what [Mypy](https://mypy.readthedocs.io/en/stable/index.html) thinks, as that's often the reasoning others use to decide when to include type hints.